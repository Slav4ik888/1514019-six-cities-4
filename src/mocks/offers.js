export default [
  {
    id: 1,
    isPremium: true,
    isFavourite: false,
    previewImage: `img/apartment-03.jpg`,
    price: 180,
    rating: 5,
    cardTitle: `Beautiful & luxurious apartment at great location`,
    offerType: `House`,
  },
  {
    id: 2,
    isPremium: true,
    isFavourite: false,
    previewImage: `img/apartment-01.jpg`,
    price: 180,
    rating: 3,
    cardTitle: `Nice, cozy, warm big bed apartment`,
    offerType: `Hotel`
  },
  {
    id: 3,
    isPremium: false,
    isFavourite: false,
    previewImage: `img/apartment-02.jpg`,
    price: 10,
    rating: 5,
    cardTitle: `Wood and stone place`,
    offerType: `Apartment`,
  },
  {
    id: 4,
    isPremium: false,
    isFavourite: false,
    previewImage: `img/apartment-01.jpg`,
    price: 180,
    rating: 5,
    cardTitle: `Beautiful & luxurious apartment at great location`,
    offerType: `Private room`,
  },
  {
    id: 5,
    isPremium: false,
    isFavourite: true,
    previewImage: `img/apartment-03.jpg`,
    price: 480,
    rating: 4,
    cardTitle: `Beautiful & luxurious apartment at great location`,
    offerType: `Hotel`
  },
  {
    id: 6,
    isPremium: false,
    isFavourite: false,
    previewImage: `img/apartment-01.jpg`,
    price: 680,
    rating: 3,
    cardTitle: `Nice, cozy, warm big bed apartment`,
    offerType: `House`,
  },
  {
    id: 7,
    isPremium: false,
    isFavourite: true,
    previewImage: `img/apartment-02.jpg`,
    price: 180,
    rating: 4,
    cardTitle: `Wood and stone place`,
    offerType: `Apartment`,
  },
  {
    id: 8,
    isPremium: true,
    isFavourite: false,
    previewImage: `img/apartment-01.jpg`,
    price: 180,
    rating: 4,
    cardTitle: `Beautiful & luxurious apartment at great location`,
    offerType: `Private room`,
  },
];
