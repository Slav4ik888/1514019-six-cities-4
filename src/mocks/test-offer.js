export const testOffer = {
  id: 1,
  isPremium: true,
  isFavourite: false,
  previewImage: `img/apartment-03.jpg`,
  pictures: [
    `img/apartment-01.jpg`,
    `img/apartment-02.jpg`,
    `img/apartment-03.jpg`,
    `img/room.jpg`,
    `img/studio-01.jpg`,
    `img/studio-photos.jpg`],
  amenities: [`Wifi`, `Heating`, `Kitchen`, `Cable TV`],
  bedrooms: 4,
  maxGuestsNumber: 5,
  description: `This cozy and complete apartment in the heart of New York is a typical, traditional and authentic feel for how New Yorkers live. I welcome all guests to stay at this place during their stay and visit of New York. 
  Please note that this apartment is on the sixth floor and there is no lift (elevator).
  The apartment is convenient for all guests, fitted with a kitchen, small dining area and two bedrooms – couples and groups of friends will find this accommodation most optimal.`,
  host: {
    photo: `img/avatar-angelina.jpg`,
    name: `Angelina Jonson`,
    super: true
  },
  price: 180,
  rating: 5,
  cardTitle: `Beautiful & luxurious apartment at great location`,
  offerType: `House`,
  coordinates: [48.856663, 2.351556],
  city: {
    zoom: 8,
    coordinates: [48.856663, 2.351556],
  },
  location: {
    zoom: 8,
  },
  reviews: [
    {
      id: 1,
      author: {
        photo: `img/avatar-angelina.jpg`,
        name: `Strange jober`
      },
      description: `small dining area and two bedrooms – couples and groups of friends will find this accommodation`,
      date: 1593095836000,
      rating: 3
    },
    {
      id: 2,
      author: {
        photo: ``,
        name: `Slava`
      },
      description: `dining area and five bedrooms – couples and groups of friends will find this accommodation`,
      date: 1593095936000,
      rating: 4.7
    },
  ],

};
