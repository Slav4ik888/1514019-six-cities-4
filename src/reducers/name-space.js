export const NameSpace = {
  DATA: `DATA`,
  TRAVEL: `TRAVEL`,
  USER: `USER`,
  FAVORITE: `FAVORITE`,
};
