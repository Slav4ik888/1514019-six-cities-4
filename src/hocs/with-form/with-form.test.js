import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';

import withForm from './with-form.js';

import {NameSpace} from '../../reducers/name-space.js';
import {offers} from '../../mocks/offers.js';
import {testOffer, reviews, nearbyOffers} from '../../mocks/test-offer.js';


const mockStore = configureStore([]);

const MockComponent = () => <div/>;

const MockComponentWrapped = withForm(MockComponent);

const authInfo = {
  avatarUrl: `/static/avatar/7.jpg`,
  email: `korzan.va@mail.ru`,
  id: 1,
  isPro: false,
  name: `Vyacheslav`,
};


it(`Render HOC <withForm/>`, () => {
  const store = mockStore({
    [NameSpace.TRAVEL]: {
      activeCity: 0,
      activeOffer: testOffer,
      activeHoverOffer: null,
    },
    [NameSpace.USER]: {
      userStatus: `AUTH`,
      authInfo,
      isLoading: false,
    },
    [NameSpace.DATA]: {
      allOffers: offers,
      comments: reviews,
      nearbyOffers,
      isLoading: false,
      review: {
        comment: ``,
        rating: null,
      },
      isError: false,
    },
  });
  const tree = renderer.create(
      <Provider store={store}>
        <MockComponentWrapped
          isDisabled={true}
          isLoading={false}
          isError={false}
          commentText={``}
          onChangeComment={() => {}}
          onChangeStars={() => {}}
          onSubmit={() => {}}
        />
      </Provider>

  ).toJSON();

  expect(tree).toMatchSnapshot();
});

// npm run test.jest -- -u with-form.test.js
// npm test with-form.test.js
