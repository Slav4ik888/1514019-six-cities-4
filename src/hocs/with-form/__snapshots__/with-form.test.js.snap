// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render HOC <withForm/> 1`] = `<div />`;
